libkineto/include/GenericTraceActivity.h

```diff
@@ -86,6 +95,9 @@ class GenericTraceActivity : public TraceActivity {
   int32_t resource{0};
   ActivityType activityType;
   std::string activityName;
+  TraceActivity* linkedAct;
+  // High-4bits represents link type.
+  uint64_t linkId;
```

Comment:
From offline discussion, how about this for now:
struct {
TraceActivity* prevActivity;
uint64_t id : 28;
uint64_t type : 4;
} flow;

prevActivity establishes order, so even if timestamps are equal we know which come first.